src/app/pages/applications/chart-releases/chart-releases.component.ts

```diff
@@ -65,21 +62,56 @@ export class ChartReleasesComponent implements OnInit {
     private sysGeneralService: SystemGeneralService) { }
 
   ngOnInit(): void {
+    this.getCatalogApps();
     this.refreshChartReleases();
-    this.refreshForms();
+    this.utils = new CommonUtils();
+
     this.refreshTable = this.modalService.refreshTable$.subscribe(() => {
       this.refreshChartReleases();
-    })
-    this.refreshForm = this.modalService.refreshForm$.subscribe(() => {
-      this.refreshForms();
     });
   }
 
-  refreshForms() {
-    this.chartReleaseForm = new ChartReleaseEditComponent(this.mdDialog,this.dialogService,this.modalService,this.appService);
-    this.plexForm = new PlexFormComponent(this.mdDialog,this.dialogService,this.modalService,this.sysGeneralService,this.appService);
-    this.nextCloudForm = new NextCloudFormComponent(this.mdDialog,this.dialogService,this.modalService,this.appService);
-    this.minioForm = new MinioFormComponent(this.mdDialog,this.dialogService,this.modalService);
+  getCatalogApps() {
+    this.appService.getAllCatalogItems().subscribe(res => {
+      if (Object.keys(res[0].trains.charts).length > 0) {
+        for (let i in res[0].trains.charts) {  // will eventually add the charts train too
+          if (i !== 'ix-chart') {
+            let item = res[0].trains.charts[i];
+            let versions = item.versions;
+            let latest, latestDetails;
+
+            let sorted_version_labels = Object.keys(versions);
+            sorted_version_labels.sort(this.utils.versionCompare);
+
+            latest = sorted_version_labels[0];
+            latestDetails = versions[latest];
+
+            switch (item.name) {
+              case 'minio':
```

Comment:
Do we still need this `switch` ?

Reply:
This PR mostly just covers the chart-release form in the UI it seems.
This Switch is just the small helptext box for the catalog items, which is not (yet) dynamic in this PR.